lua/neogit.lua

```diff
@@ -31,16 +31,39 @@ local setup = function(opts)
     config.values = vim.tbl_deep_extend("force", config.values, opts)
   end
 
-  if config.values.use_magit_keybindings then
-    config.values.mappings.status["F"] = "PullPopup"
-    config.values.mappings.status["p"] = ""
-  end
-
-  hl.setup()
-  signs.setup()
-  state.setup()
+  local config_errs = config.validate_config()
+  if vim.tbl_count(config_errs) > 0 then
+    local header = "====Neogit Configuration Errors===="
+    local header_message = {
+      "Neogit has NOT been setup!",
+      "You have a misconfiguration in your Neogit setup!",
+      'Validate that your configuration passed to `require("neogit").setup()` is valid!',
+    }
+    local header_sep = ""
+    for _ = 0, string.len(header), 1 do
+      header_sep = header_sep .. "-"
+    end
 
-  require("neogit.autocmds").setup()
+    local config_errs_message = {}
+    for config_key, err in pairs(config_errs) do
+      table.insert(config_errs_message, string.format("Config value: `%s` had error -> %s", config_key, err))
+    end
+    error(
+      string.format(
+        "\n%s\n%s\n%s\n%s",
+        header,
+        table.concat(header_message, "\n"),
+        header_sep,
+        table.concat(config_errs_message, "\n")
+      ),
+      vim.log.levels.ERROR
+    )
```

Comment:
Following the `module.setup()` pattern, can this get extracted to `config.setup()`? I don't think the validation logic needs to live outside the module it applies to.

Reply:
Yeah, that's fair. I'll move it :slightly_smiling_face: